SharpMap/CoordinateSystems/Transformations/GeometryTransform.cs

```diff
@@ -36,8 +36,11 @@ public static Envelope TransformBox(Envelope box, IMathTransform transform)
         {
             if (box == null)
                 return null;
-            
-            var corners = new [] {
+
+            if (box.IsNull)

```

Comment:
I'd prefer to return `new Envelope(box)` instead of `null`

Reply:
Done